### PR TITLE
feat(services-auth-api): Create scope for passkeys

### PR DESCRIPTION
**Pull request by baering**
## What

Create a new scope `@island.is/auth/passkeys` and require it for passkey endpoints in `public-api` and `ids-api`.

## Why

To ensure only those with the scope can access this feature

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Enhanced authorization scope handling with the introduction of a new `passkeys` scope for improved security and access management.
  
- **Improvements**
  - Updated the passkeys module to include new authorization scopes, enhancing the security measures for passkey-related endpoints.

### CODE REVIEW

File: libs/auth-api-lib/seeders/data/scope-passkeys.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { compose, createScope } from './helpers'
```
[the remark this answers is not in the document]
Comment (baering) [reply libs/auth-api-lib/seeders/data/scope-passkeys.ts:1]: It is used in line 3 from col 19.

File: libs/auth-api-lib/seeders/data/scope-passkeys.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export const up = compose(
+  createScope({
+    name: '@island.is/auth/passkeys',
+    displayName: 'Aðgangslyklar',
+    description:
+      'Gerir notanda kleift að innskrá sig með lífkenni í gegnum aðgangslykil',
+    addToResource: '@island.is',
+    addToClients: ['@island.is/app', '@island.is/web'],
+    delegation: {
```
Comment (Herdismaria) [inline libs/auth-api-lib/seeders/data/scope-passkeys.ts:11]: We probably need to add entries to the `api_scope_delegation_types` table when creating scopes this way.

**Review by Herdismaria — CHANGES_REQUESTED** [review]
One thing about creating scopes with migration. The same comment could be said if we are creating clients this way and adding delegations to them. Then we need to add entries to the `client_delegation_types`.

**Review by Herdismaria — APPROVED** [review]
⚡

**Review by saevarma — APPROVED** [review]

**Pull request merged**